envy.sh

```diff
@@ -20,37 +79,45 @@ envy() {
         CONTENTS=$(cat "${FILENAME}" | grep "^[^#]")
     fi
     while read -r PAIR; do
-        K=$(echo ${PAIR} | sed 's/\([^=]*\)=\(.*\)/\1/')
-        V=$(echo ${PAIR} | sed 's/\([^=]*\)=\(.*\)/\2/')
+        K=$(sed 's/\([^=]*\)=\(.*\)/\1/' <<< "${PAIR}")
+        V=$(sed 's/\([^=]*\)=\(.*\)/\2/' <<< "${PAIR}")
+        # Check if templating
+        if grep -q "{{.*}}" <<< "${V}"; then
+            V=$(mush <<< "${V}")
+        fi
         if grep -q "^_INCLUDE" <<< "${K}"; then
-            envy "${V}"
+            process_input "${V}"
         else
-            # If variable not already set then export
-            if [ -z "$(printenv ${K})" ]; then
-                BASH_ESCAPED_VALUE=$(sed 's/\([$\\ ]\)/\\\1/g' <<< "${V}")
-                BASH_FORMAT="export ${K}=${BASH_ESCAPED_VALUE}"
-                eval "${BASH_FORMAT}"
-                if [ "${OUTPUT}" == "bash" ]; then
-                    echo "${BASH_FORMAT}"
-                elif [ "${OUTPUT}" == "env-file" ]; then
-                    echo "${PAIR}"
-                elif [ "${OUTPUT}" == "make" ]; then
-                    MAKE_ESCAPED_VALUE=$(sed 's/\([$]\)/$\1/g' <<< "${V}")
-                    MAKE_ESCAPED_VALUE=$(sed 's/\([#\\]\)/\\\1/g' <<< "${MAKE_ESCAPED_VALUE}")
-                    echo "export ${K}=${MAKE_ESCAPED_VALUE}"
-                fi
-            fi
+            export ${K}="$(bash_escape "${V}")"
```

Comment:
Thought this should be failing given the 2 sets of double quotes, are we not exercising this in tests?

Reply:
It's definitely exercised and seems to be parsed correctly by bash, i guess the $() is evaluated as a separate expression since its in the subshell